assets/js/modules/adsense/components/dashboard/AdBlockingRecoveryWidget.test.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * AdBlockingRecoveryWidget component tests.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import AdBlockingRecoveryWidget from './AdBlockingRecoveryWidget';
+import {
+	act,
+	fireEvent,
+	render,
+	createTestRegistry,
+	provideSiteInfo,
+	provideUserAuthentication,
+	provideModules,
+	unsubscribeFromAll,
+} from '../../../../../../tests/js/test-utils';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	MODULES_ADSENSE,
+} from '../../datastore/constants';
+import {
+	ACCOUNT_STATUS_PENDING,
+	ACCOUNT_STATUS_READY,
+	SITE_STATUS_ADDED,
+	SITE_STATUS_READY,
+} from '../../util';
+import { getWidgetComponentProps } from '../../../../googlesitekit/widgets/util';
+import { WEEK_IN_SECONDS, DAY_IN_SECONDS } from '../../../../util';
+
+describe( 'AdSenseConnectCTA', () => {
+	let registry;
+	const timestampThreeWeeksAgo = Date.now() / 1000 - WEEK_IN_SECONDS * 3;
```

Comment:
I was trying to speed things up here to avoid another round of CR/Execution by changing this to a value in seconds, but unfortunately one of the tests is now failing - did I misunderstand the intent of this value?

Also I realise this is now potentially a float, and should have `Math.floor()` applied:

```suggestion
	const timestampThreeWeeksAgo = Math.floor( Date.now() / 1000 ) - WEEK_IN_SECONDS * 3;
```

Reply:
The previous implementation of `timestampLessThanThreeWeeksAgo` calculates the timestamp for a day less than `timestampThreeWeeksAgo`, which means it's about three weeks and one day ago from the current time, not "less than three weeks ago".

So, to calculate the timestamp that is less than three weeks ago, we should add time (in seconds) instead of subtracting like so:

`const timestampLessThanThreeWeeksAgo = timestampThreeWeeksAgo + HOUR_IN_SECONDS;`

Reply:
Great, thanks for clearing that up!